Adding RHOMBIO_L476DMW1K as a new Board target

guialonsoalb opened the pull request:
### Description

This pull request contains files necessary to the execution of Mbed-OS over RHOMBIO_L476DMW1K IoT Kit. The kit is based on STM32L476VG and the files to be added are a porting based on target DISCO_L476VG, sharing most of its pin and peripheral configuration.

The kit has been tested with ARM, GCC_ARM and IAR.

Greentea logs attached below:
[RHOMBIO_L476DMW1K-GCC_ARM-TESTS-PASS.txt](https://github.com/ARMmbed/mbed-os/files/2886419/RHOMBIO_L476DMW1K-GCC_ARM-TESTS-PASS.txt)
[RHOMBIO_L476DMW1K-ARM-TESTS-PASS.txt](https://github.com/ARMmbed/mbed-os/files/2886420/RHOMBIO_L476DMW1K-ARM-TESTS-PASS.txt)
[RHOMBIO_L476DMW1K-IAR-TESTS-PASS.txt](https://github.com/ARMmbed/mbed-os/files/2886421/RHOMBIO_L476DMW1K-IAR-TESTS-PASS.txt)

### Pull request type

    [ ] Fix
    [ ] Refactor
    [x] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers

@MarceloSalazar 

### Release Notes

guialonsoalb committed:
Added new target RHOMBIO_L476DMW1K

guialonsoalb committed:
added licence identifier texts

cmonr commented on targets/TARGET_STM/TARGET_STM32L4/TARGET_STM32L476xG/TARGET_RHOMBIO_L476DMW1K/PeripheralNames.h:
```diff
@@ -0,0 +1,100 @@
+    LPUART_1 = (int)LPUART1_BASE
+} UARTName;
+
+//#define STDIO_UART_TX  PA_2
```
Any particular reason this is commented out instead of removed?

guialonsoalb replied:
It can be removed. Wasn't sure if it could be needed in the future

cmonr commented on targets/TARGET_STM/TARGET_STM32L4/TARGET_STM32L476xG/TARGET_RHOMBIO_L476DMW1K/PinNames.h:
```diff
@@ -0,0 +1,375 @@
+    ADC_VBAT = 0xF2,
+
+    // STDIO for console print
+/*#ifdef MBED_CONF_TARGET_STDIO_UART_TX
```
Is this still needed for some reason?

guialonsoalb replied:
Same here, It can be removed.

0xc0170 replied:
Please remove not used code.

cmonr reviewed (commented):
LGTM, just some stale comment nits.

0xc0170 commented on targets/TARGET_STM/TARGET_STM32L4/TARGET_STM32L476xG/TARGET_RHOMBIO_L476DMW1K/PinNames.h:
```diff
@@ -0,0 +1,375 @@
+/* mbed Microcontroller Library
+
+* SPDX-License-Identifier: Apache-2.0
```
SPDX is wrong in some files, the below is BSD-style license but SPDX states Apache 2.0? Please fix

0xc0170 commented on targets/TARGET_STM/TARGET_STM32L4/TARGET_STM32L476xG/TARGET_RHOMBIO_L476DMW1K/system_clock.c:
```diff
@@ -0,0 +1,370 @@
+/* mbed Microcontroller Library
+* Copyright (c) 2006-2017 ARM Limited
```
here SPDX Apache 2.0 identifier

guialonsoalb committed:
Change in license identifier, removed commented dead code

Change in license identifier
Removed commented dead code
Removed some tab space
